### PR TITLE
Allow creation of "Asides" #20

### PR DESCRIPTION
**Pull request by YorgunKelebek**
Addressing issue #20 Allow creation of "Asides"

### CODE REVIEW

**Comment by YorgunKelebek** [comment]
image

File: css/app.css [hunk 1]
```diff
@@ -87,6 +87,21 @@ img.blog-image + div.summary-blog-section {
 .blog-preview-close:after {
     content: "Close";
 }
+
+aside.aside-note {
+    width: 40%;
```
Comment (goofballLogic) [inline css/app.css]: how will this work on mobile?

File: js/app.js [hunk 2]
```diff
@@ -197,29 +197,32 @@ function tagIsSelected(blogTag) {
...
-                }
-            }
-            if (data[key].system.type === "code_snippet") {
-                var objCodeSnippet = blogSummary.querySelector("[data-codename='" + key + "']");
-                if (objCodeSnippet !== null) {
-                    var elementCodeSnippet = document.createElement('pre');
-                    elementCodeSnippet.textContent = data[key].elements.code.value;
-                    objCodeSnippet.parentNode.replaceChild(elementCodeSnippet, objCodeSnippet);
-                }
+            if (snippetTypes.hasOwnProperty(data[key].system.type)) {
```
Comment (goofballLogic) [inline js/app.js]: the `in` keyword would be sufficient here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in

File: js/app.js [hunk 3]
```diff
@@ -197,29 +197,32 @@ function tagIsSelected(blogTag) {
...
         }
     }
 }
+function buildContentSnippet(blogSummary, data, key) {
+    const snippetType = snippetTypes[data[key].system.type];
+    const objSnippet = blogSummary.querySelector("[data-codename='" + key + "']");
+    if (objSnippet !== null) {
+        const elementSnippet = document.createElement(snippetType.tag);
+        if (snippetType.format === "text") elementSnippet.textContent = firstOrDefaultValue(data[key], snippetType.property) || "";
+        else elementSnippet.innerHTML = firstOrDefaultValue(data[key], snippetType.property) || "";
```
Comment (goofballLogic) [inline js/app.js]: repeated code - can DRY

File: js/app.js (hunk 2 above)
Comment (YorgunKelebek) [reply js/app.js]: Ah, thanks!

File: js/app.js (hunk 3 above)
Comment (YorgunKelebek) [reply js/app.js]: Ironic, seeing as the whole point of this refactoring was to DRY!

**Comment by YorgunKelebek** [comment]
> I'm interested to see how this works for the aside I have in my blog post which contains both a heading and italic formatting

Looks a little odd due to the image below, but you can move the aside above the paragraphs before it to re-position.

image

**Comment by goofballLogic** [comment]
Shouldn't the image respect the aside also?
And, none of my formatting in the Aside shows up.

⁣Sent from BlueMail ​


-------- Original Message --------

File: css/app.css [hunk 4]
```diff
@@ -89,19 +90,26 @@ img.blog-image + div.summary-blog-section {
...
+        width: 40%;
+    }
+}
 
 aside.aside-note > p {
     margin: .125rem;
 }
 
+aside.aside-note > h1, h2, h3, h4, h5, h6 {
+    font-size: larger;
```
Comment (goofballLogic) [inline css/app.css]: Does this mean that heading size in asides is ignored? I suppose that could be valid.

File: js/app.js [hunk 5]
```diff
@@ -217,8 +217,9 @@ function buildContentSnippet(blogSummary, data, key) {
     const objSnippet = blogSummary.querySelector("[data-codename='" + key + "']");
     if (objSnippet !== null) {
         const elementSnippet = document.createElement(snippetType.tag);
-        if (snippetType.format === "text") elementSnippet.textContent = firstOrDefaultValue(data[key], snippetType.property) || "";
-        else elementSnippet.innerHTML = firstOrDefaultValue(data[key], snippetType.property) || "";
+        const snippetContent = firstOrDefaultValue(data[key], snippetType.property) || "";
+        if (snippetType.format === "text") elementSnippet.textContent = snippetContent;
+        else elementSnippet.innerHTML = snippetContent;
```
Comment (goofballLogic) [inline js/app.js]: Some people prefer a ternary expression here but I think it's down to personal preference. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

File: css/app.css (hunk 4 above)
Comment (YorgunKelebek) [reply css/app.css]: Yes, I don't think different heading styles should be allowed on the aside snippet

**Comment by YorgunKelebek** [comment]
> Shouldn't the image respect the aside also? And, none of my formatting in the Aside shows up. 

Not sure about the image, it can't shrink I guess so gets pushed down...
I will remove the formatting from the aside class (italics, color etc.)

**Pull request merged**